### PR TITLE
fix: Enable OIDC with Synology SSO Server

### LINKED ISSUE #4543: [BUG] - OIDC with Synology not working
### First Check

- [X] This is not a feature request.
- [X] I added a very descriptive title to this issue (title field is above this).
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.
- [ ] This issue can be replicated on the demo site (https://demo.mealie.io/).

### What is the issue you are experiencing?

I Configured the SSO Server on my Synology, but the Loging won't function.

Mealie requires the `name` claim from an OIDC Provider, but the Synology SSO Server doesn't Provide this claim. And to my knowlegde this isn't configurable.

Would it be possible to not require `name` as a claim from the OIDC Provider? On a first glance it doesn't seam required.

Maybe it would be possible to set Full Name equal to username if no `name` claim is present. For the `username` claim there is something similar configured: https://github.com/mealie-recipes/mealie/blob/622c1b11f5f84f73529e0c5567d34f11f9685c79/mealie/core/security/providers/openid_provider.py#L64C1-L77C38

### Steps to Reproduce

1) Configure Synology SSO Server
2) Try to Loging via SSO

### Please provide relevant logs

[OIDC] Required claims not present. Expected: {'email', 'name'} Actual: dict_keys(['aud', 'auth_time', 'email', 'exp', 'iat', 'iss', 'nonce', 'sub', 'username'])

### Mealie Version

2.2.0

### Deployment

Docker (Synology)

### Additional Deployment Details

_No response_

### PR DESCRIPTION
**Pull request by JoTec2002**
## What type of PR is this?

_(REQUIRED)_

- feature
- bug

## What this PR does / why we need it:

Mealie required the `name` claim from an OIDC Provider, but the Synology SSO Server doesn't Provide this claim.

After this Pull Request Mealie will no longer required the `name` claim to be present.

To not cause any issues or lose any OIDC user information the following logic is used:
If the `name` claim is present it will be used, otherwise the username will be used as name.

## Which issue(s) this PR fixes:

Fixes #4543

### CODE REVIEW

**Comment by MuadDibVV** [comment]
+1

**Comment by cmintey** [comment]
I think it would be better and more extensible to introduce a new env var `OIDC_NAME_CLAIM`, which would default to `name`, but you could override it to whatever you want.

**Comment by michael-genson** [comment]
@JoTec2002 if you could take a look at the feedback from @cmintey we should be able to get this merged!

**Comment by JoTec2002** [comment]
@cmintey I completed the changes you suggested.

File: mealie/core/security/providers/openid_provider.py [hunk 1]
```diff
@@ -96,7 +98,7 @@ def authenticate(self) -> tuple[str, timedelta] | None:
     def required_claims(self):
         settings = get_app_settings()
 
-        claims = {"name", "email", settings.OIDC_USER_CLAIM}
+        claims = {"email", settings.OIDC_USER_CLAIM}
```
Comment (cmintey) [inline mealie/core/security/providers/openid_provider.py]: Can you add the new env var to this set, since it's required?

**Review by cmintey — CHANGES_REQUESTED** [review]
Looking good! One more small thing to address

File: mealie/core/security/providers/openid_provider.py (hunk 1 above)
Comment (JoTec2002) [reply mealie/core/security/providers/openid_provider.py]: Done

**Review by cmintey — APPROVED** [review]
Thanks for working on this!

**Review by michael-genson — APPROVED** [review]

**Pull request merged**